src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -100,6 +116,28 @@
   private final ShardBackplane backplane;
   private final LoadingCache<String, Instance> workerStubs;
 
+  class LocalCasWriter implements CasWriter {
```

Comment:
This feels awkward - with the only caller of insertFile/BlobLocally and insertFile/BlobToCasMember being these classes, they should retain those methods (and their helpers), rather than the outer class.